docs/serving/samples/hello-world/helloworld-csharp/Dockerfile

```diff
@@ -1,18 +1,20 @@
-# Use Microsoft's official .NET image.
+# Use Microsoft's official lightweight .NET images.
 # https://hub.docker.com/r/microsoft/dotnet
-FROM microsoft/dotnet:2.2-sdk
 
-# Install production dependencies.
-# Copy csproj and restore as distinct layers.
+FROM microsoft/dotnet:2.2-sdk-alpine AS build
 WORKDIR /app
-COPY *.csproj .
-RUN dotnet restore
 
-# Copy local code to the container image.
-COPY . .
+# Copy csproj and restore as distinct layers.
+COPY *.csproj ./
+RUN dotnet restore
 
-# Build a release artifact.
+# Copy and publish app.
+COPY . ./
 RUN dotnet publish -c Release -o out
 
-# Run the web service on container startup.
-CMD ["dotnet", "out/helloworld-csharp.dll"]
+# Run the web service on container startup in a lean production image.
+FROM microsoft/dotnet:2.2-aspnetcore-runtime-alpine
+WORKDIR /app
+
+COPY --from=build /app/out ./
+CMD ["dotnet", "helloworld-csharp.dll"]
```

Comment:
I chose not to use alpine because using alpine is generally an informed decision teams make if they're okay with some narrow use case limitations it has. I figured docs would be best to use the "vanilla" version of images recommended by the docs from these different platforms. For example, in my PR that @grayside linked to, I have a Dockerfile that looks basically like this, since that's how their official docs recommend doing it, except I don't use alpine.

Reply:
Sure, feel free to remove C# from this PR.

Reply:
Updated the PR without C#. Resolving C# conversations...